source/OptimaJet.DWKit.StarterApplication/Services/Auth0ManagementApiTokenService.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using static OptimaJet.DWKit.StarterApplication.Utility.EnvironmentHelpers;
+using RestSharp;
+using Newtonsoft.Json.Linq;
+using Serilog;
+
+namespace OptimaJet.DWKit.StarterApplication.Services
+{
+    public class Auth0ManagementApiTokenService
+    {
+        private string token;
+        public Auth0ManagementApiTokenService()
+        {
+        }
+
+        public string Token
+        {
+            get
+            {
+                if (token == null) {
+                    var domainUrl = GetVarOrThrow("AUTH0_DOMAIN");
+                    var clientId = GetVarOrDefault("AUTH0_TOKEN_ACCESS_CLIENT_ID", "");
+                    var clientSecret = GetVarOrDefault("AUTH0_TOKEN_ACCESS_CLIENT_SECRET", "");
+                    var client = new RestClient($"{domainUrl}/oauth/token");
+                    var request = new RestRequest(Method.POST);
+                    request.AddHeader("content-type", "application/json");
+                    request.AddParameter("application/json", $"{{\"grant_type\":\"client_credentials\",\"client_id\": \"{clientId}\",\"client_secret\": \"{clientSecret}\",\"audience\": \"{domainUrl}/api/v2/\"}}", ParameterType.RequestBody);
+                    IRestResponse response = client.Execute(request);
+                    if (response.IsSuccessful)
+                    {
+                        dynamic json = JObject.Parse(response.Content);
+                        token = json.access_token;
+                    }
+                    else 
+                    {
+                        Log.Error($"Failed to request token from Auth0: Status={response.StatusDescription}, Content={response.Content}");
```

Comment:
for whomever is consuming this property, how will they know if token renewal fails?

Reply:
I wasn't quite sure how to handle this.  It will silently fail to update the user's GivenName and FamilyName.  We added exception handling in HttpCurrentUserContext and it will be logged.

Reply:
In my experience silent failure makes tracking things harder.
Logging isn't bad, but something should be done with the logs. 
like, have you used rollbar? or some other exception / error / log tracking service?

Reply:
I have used log entries, but wasn't a fan of it.  We should have some service to collect logs.  

This one is so low value, I am not worried about it.